test/server/http_filter_base_test.cc

```diff
@@ -90,5 +90,29 @@ TEST_P(HttpFilterBaseIntegrationTest, EmptyRequestLevelConfigurationShouldFail)
   EXPECT_THAT(response->body(), HasSubstr(kBadConfigErrorSentinel));
 }
 
+TEST_P(HttpFilterBaseIntegrationTest, MultipleValidConfigurationHeadersFails) {
+  // Make sure we fail when two valid configuration headers are send.
+  setRequestLevelConfiguration("{}");
+  appendRequestLevelConfiguration("{}");
+  Envoy::IntegrationStreamDecoderPtr response = getResponse(ResponseOrigin::EXTENSION);
+  ASSERT_TRUE(response->complete());
+  EXPECT_THAT(response->body(),
+              HasSubstr("Received multiple configuration headers in the request"));
+}
+
+TEST_P(HttpFilterBaseIntegrationTest, SingleValidPlusEmptyConfigurationHeadersFails) {
+  // Make sure we fail when both a valid configuration plus an empty configuration header is send.
+  // Note that we could be more flexible and look for the first request header that has a value,
```

Comment:
This note seems like it belongs more in the function itself rather than in this test.

Also, I might encourage a rephrase of why our approach is reasonable.  Without understanding of a real use case for them, we are assuming that any existence of duplicate headers here is in error.

Reply:
Amended this; also updated the Envoy dep to f95f5391b0b8683081ec786ea946026594955fc6 in here, as that didn't need require any changes on our side modulo updating the revision/sha in `.bazelrc`.

Reply:
(changes associated to your review comment applied in 6136e81, the Envoy sha change is in 30d7f20)